### PR TITLE
Missing cppyy headers

### LINKED ISSUE #84: how to construct iet over number field in pyintervalxt?
```
import cppyy
from pyeantic import eantic
from pyintervalxt import intervalxt

L = eantic.renf("a^3 - a^2 - a - 1", "a", "[1.84 +/- 0.01]")
lengths = []
lengths.append(eantic.renf_elem(L, "a"))
lengths.append(eantic.renf_elem(L, [int(1), int(2), int(-3)]))
DTYPE_T = cppyy.gbl.eantic.renf_elem_class
lengths = intervalxt.sample.Lengths[DTYPE_T](lengths)
iet = intervalxt.makeIET(lengths, [int(1), int(0)])
```
results in
```
ValueError: Template method resolution failed:
  none of the 6 overloaded methods succeeded. Full details:
  intervalxt::IntervalExchangeTransformation intervalxt::makeIET(const intervalxt::sample::Lengths<eantic::renf_elem_class>& lengths, vector<int> permutation) =>
    ValueError: nullptr result where temporary expected
  intervalxt::IntervalExchangeTransformation intervalxt::makeIET(intervalxt::sample::Lengths<eantic::renf_elem_class>*const& lengths, vector<int> permutation) =>
    ValueError: nullptr result where temporary expected
  intervalxt::IntervalExchangeTransformation intervalxt::makeIET(const intervalxt::sample::Lengths<eantic::renf_elem_class>& lengths, vector<int> permutation) =>
    ValueError: nullptr result where temporary expected
  intervalxt::IntervalExchangeTransformation intervalxt::makeIET(const intervalxt::sample::Lengths<eantic::renf_elem_class>& lengths, vector<int> permutation) =>
    ValueError: nullptr result where temporary expected
  intervalxt::IntervalExchangeTransformation intervalxt::makeIET(intervalxt::sample::Lengths<eantic::renf_elem_class>*const& lengths, vector<int> permutation) =>
    ValueError: nullptr result where temporary expected
  intervalxt::IntervalExchangeTransformation intervalxt::makeIET(const intervalxt::sample::Lengths<eantic::renf_elem_class>& lengths, vector<int> permutation) =>
    ValueError: nullptr result where temporary expected
  intervalxt::IntervalExchangeTransformation intervalxt::makeIET(const intervalxt::sample::Lengths<eantic::renf_elem_class>& lengths, vector<int> permutation) =>
    ValueError: nullptr result where temporary expected
  intervalxt::IntervalExchangeTransformation intervalxt::makeIET(intervalxt::sample::Lengths<eantic::renf_elem_class>*const& lengths, vector<int> permutation) =>
    ValueError: nullptr result where temporary expected
  intervalxt::IntervalExchangeTransformation intervalxt::makeIET(const intervalxt::sample::Lengths<eantic::renf_elem_class>& lengths, vector<int> permutation) =>
    ValueError: nullptr result where temporary expected
```

### PR DESCRIPTION
**Pull request by videlec**
Fixes #84, #91, #92.

Helps for #79.

### CODE REVIEW

File: pyintervalxt/test/interval_exchange_transformation.py [hunk 1]
```diff
@@ -56,4 +56,36 @@ def test_reduce():
     assert repr(iet2) == "[c: 23] [d: 11] [e: 21] / [e] [d] [c]"
     assert repr(iet3) == "[f: 9] [g: 65] / [g] [f]"
 
+def iet_check(iet):
+    saf = iet.safInvariant()
+    decomposition = intervalxt.DynamicalDecomposition(iet)
+    decomposition.decompose()
+    ncyls = sum(bool(component.cylinder() == True) for component in decomposition.components())
```
Comment (saraedum) [inline pyintervalxt/test/interval_exchange_transformation.py]: Did you mean to check something about `ncyls`?

File: pyintervalxt/test/interval_exchange_transformation.py [hunk 2]
```diff
@@ -56,4 +56,36 @@ def test_reduce():
...
+
+def test_mpq_constructor():
+    from gmpxxyy import mpq
+    from pyintervalxt import intervalxt
+    lengths = intervalxt.sample.Lengths[mpq]([1, 1])
+    iet = intervalxt.makeIET(lengths, [1, 0])
+    iet_check(iet)
+
+def test_eantic_constructor():
+    from pyintervalxt import intervalxt
```
Comment (saraedum) [inline pyintervalxt/test/interval_exchange_transformation.py]:
```suggestion
    lengths = intervalxt.sample.Lengths[mpz]([1, 1])
    iet = intervalxt.makeIET(lengths, [1, 0])
    iet_check(iet)

def test_mpq_constructor():
    from gmpxxyy import mpq
    lengths = intervalxt.sample.Lengths[mpq]([1, 1])
    iet = intervalxt.makeIET(lengths, [1, 0])
    iet_check(iet)

def test_eantic_constructor():
```

File: pyintervalxt/test/interval_exchange_transformation.py (hunk 1 above)
Comment (videlec) [reply pyintervalxt/test/interval_exchange_transformation.py]: Not really. Just that it works

Comment (videlec) [reply pyintervalxt/test/interval_exchange_transformation.py]: I added a bit more.

**Pull request merged**